transformer_sidecar/src/transformer_sidecar/transformer.py

```diff
@@ -154,6 +154,43 @@ def prepend_xcache(file_paths):
         prefixed_paths.append(f'root://{prefix_list[pinned_xcache_index]}//{f}')
     return prefixed_paths
 
+
+def custom_path_sorting(file_paths):
+
+    preferred = os.environ.get('PREFERRED_ENDPOINTS', '')
+    avoided = os.environ.get('AVOIDED_ENDPOINTS', '')
+
+    if not preferred and not avoided:
+        return file_paths
+
+    sorted_paths = []
+    unsorted_paths = []
+    preferred_list = []
+    avoided_list = []
+
+    if preferred:
+        preferred_list = [p.strip() for p in preferred.split(',')]
+
+    if avoided:
+        avoided_list = [p.strip() for p in avoided.split(',')]
+
+    for f in file_paths:
+        skip = False
+        for av in avoided_list:
```

Comment:
The avoided list could be implemented more cleanly with the python `filter` function:
```python
acceptable_paths = list(filter(lambda x: x not in avoided_list, file_paths))
```

Reply:
Or list comprehension ;-) :

```
acceptable_paths = [x for x in file_paths if not any(a in x for a in avoided_list)]
```

Reply:
I am sorry for offending your coding styles. I will delete this PR.